homeassistant/components/emulated_hue/upnp.py

```diff
@@ -60,13 +68,26 @@ class UPNPResponderThread(threading.Thread):
 
     _interrupted = False
 
-    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast):
+    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast,
+                 advertise_ip, advertise_port):
         """Initialize the class."""
         threading.Thread.__init__(self)
 
         self.host_ip_addr = host_ip_addr
         self.listen_port = listen_port
         self.upnp_bind_multicast = upnp_bind_multicast
+        self.advertise_ip = advertise_ip
+        self.advertise_port = advertise_port
+
+        # If using the advertisement overides, only
+        # use them for forming the responses
+        resp_ip = self.advertise_ip if \
```

Comment:
Thinking this through -- I can move the logic into the config class itself, and since that's already being passed around, I think that provides the cleanest interface here.

Will take a look at why those unit tests failed in CI -- tox ran clean for me initially.